### ISSUE TITLE
Decide what type of training we want to apply in RNN neuron modules

### ISSUE THREAD

**Issue by dbeinhauer** [dbeinhauer/mcs-source#85]
We need to decide whether we want to remember the gradients of the hidden states. If so, we need to tackle with the multiple backward calling (probably will need to use something different than retain_graph=True as it is very memory demanding. This approach would be good in case we want to design some loss function for the LSTM block itself.

If we do not need the gradients, we just want to detach the gradients from the hidden states (to get over the problem with multiple backward callings).

**Comment by dbeinhauer** [comment]
We decided to use the detach variant. Currently, the model back-propagates only one time step backward. It seems to work fine and the other variants would be technically difficult and much more memory consuming.
